plexus-compilers/plexus-compiler-javac/src/main/java/org/codehaus/plexus/compiler/javac/JavacCompiler.java

```diff
@@ -128,6 +137,44 @@ public String getCompilerId() {
         return "javac";
     }
 
+    private String getInProcessJavacVersion() throws CompilerException {
+        return System.getProperty("java.version");
+    }
+
+    private String getOutOfProcessJavacVersion(String executable) throws CompilerException {
+        String version = VERSION_PER_EXECUTABLE.get(executable);
```

Comment:
`computeIfAbsent` ?

Reply:
tricky because of the checked exception which might be thrown there.

Reply:
so use simple HashMap and synchronize access in code

Reply:
possible, not sure if the additional complexity is worth it. Do you have a strong opinion about it?

Reply:
> possible, not sure if the additional complexity is worth it. Do you have a strong opinion about it?

Ok, can be simple, at most there will be a few execution for version